lua/test.sh

```diff
@@ -28,15 +28,18 @@ for t in $test_files; do
 	t=${t%.lua}
 	t=${t#./}
 	printf "%-30s" "$t"
-	$VIS "$t.in" < /dev/null 2> /dev/null > "$t.busted"
-
+	mkfifo infifo
+	$VIS "$t.in" <infifo 2> /dev/null > "$t.busted" &
+	for i in 1;	do sleep 0.1s; echo ":qall!"; done > infifo &
```

Comment:
Not a fan of the non-posix sleep usage here and in the lua script.

Reply:
> `read -t 0.1`

Also not defined in posix. Its not elegant but to keep the script completely posix compatible and minimize the sleep cost something like this should work:
```sh
	if [ $(grep -q "vis:communicate" < "$t.lua") ]; then
		{ sleep 1; echo ":qall!"; } > infifo &
	else
		echo ":qall!" > infifo &
	fi
```

Reply:
What do you mean by "non-posix" and "not defined in posix"? I checked it on wiki and found both commands in the list of posix commands.
https://en.wikipedia.org/wiki/List_of_POSIX_commands

There is a problem with sleep in the shell script, because it is a bad way to overcome a race condition. We need to give async processes some time to finish and produce all necessary events, but there should be a better way for syncronization and I am still thinking about it.

Reply:
> What do you mean by "non-posix" and "not defined in posix"? I checked it on wiki and found both commands in the list of posix commands. https://en.wikipedia.org/wiki/List_of_POSIX_commands

https://pubs.opengroup.org/onlinepubs/9699919799/